packages/block-editor/src/hooks/style.js

```diff
@@ -120,24 +112,27 @@ export function getInlineStyles( styles = {} ) {
 }
 
 function compileElementsStyles( selector, elements = {} ) {
-	return map( elements, ( styles, element ) => {
-		const elementStyles = getInlineStyles( styles );
-		if ( ! isEmpty( elementStyles ) ) {
-			// The .editor-styles-wrapper selector is required on elements styles. As it is
-			// added to all other editor styles, not providing it causes reset and global
-			// styles to override element styles because of higher specificity.
-			return [
-				`.editor-styles-wrapper .${ selector } ${ ELEMENTS[ element ] }{`,
-				...map(
-					elementStyles,
-					( value, property ) =>
-						`\t${ kebabCase( property ) }: ${ value };`
-				),
-				'}',
-			].join( '\n' );
-		}
-		return '';
-	} ).join( '\n' );
+	return Object.keys( elements )
+		.map( ( key ) => {
+			const elementStyles = getInlineStyles( elements[ key ] );
+			if ( ! isEmpty( elementStyles ) ) {
+				// The .editor-styles-wrapper selector is required on elements styles. As it is
+				// added to all other editor styles, not providing it causes reset and global
+				// styles to override element styles because of higher specificity.
+				return [
+					`.editor-styles-wrapper .${ selector } ${ ELEMENTS[ key ] }{`,
+					...Object.keys( elementStyles ).map(
```

Comment:
Same here regarding `Object.keys` vs `entries`.

Also if we pulled Object.keys/entries above isEmpty we could get rid of isEmpty from here as well - Though not sure if we have a long term goal of removing dependency on lodash altogether :)

Reply:
👍 

> Also if we pulled Object.keys/entries above isEmpty we could get rid of isEmpty from here as well

Good point

I'm working on `getInlineStyles` for another, unrelated task so I might postpone removing `isEmpty` until that's finalized. 

> Though not sure if we have a long term goal of removing dependency on lodash altogether

I don't think there's a push to remove it entirely - it's pretty useful in a lot of situations.  

I think I've read that it's encouraged anywhere there's a vanilla alternative that makes sense and isn't too costly in terms of code footprint. I went looking for the comment but can't find it, sorry!

Reply:
no worries and thanks for the insights!